### PR TITLE
Fixup View constructor death tests

### PR DESCRIPTION
**Pull request by dalg24**
Instead of guarding the content of the test and give the false impression that the tests ran, use GTest native support to signify that the tests are not run

### CODE REVIEW

File: core/unit_test/TestViewCtorDimMatch.hpp [hunk 1]
```diff
@@ -193,9 +202,9 @@ TEST(TEST_CATEGORY_DEATH, view_construction_with_wrong_static_extents) {
   CHECK_DEATH_UNMANAGED({ Kokkos::View<int[1][1][1][1][1][1][1]>    v(nullptr, 2, 1, 1, 1, 1, 1, 1); });
   CHECK_DEATH_UNMANAGED({ Kokkos::View<int[1][1][1][1][1][1][1][1]> v(nullptr, 2, 1, 1, 1, 1, 1, 1, 1); });
   // clang-format on
-#endif
 }
 
 #undef CHECK_DEATH
+#undef CHECK_DEATH_UNMANAGED
```
Comment (crtrott) [inline core/unit_test/TestViewCtorDimMatch.hpp:208]: drive-by fix is fine with me.

**Review by crtrott — APPROVED** [review]

File: core/unit_test/TestViewCtorDimMatch.hpp [hunk 2]
```diff
@@ -93,6 +93,9 @@ TEST(TEST_CATEGORY_DEATH, view_construction_with_wrong_params_dyn) {
   ::testing::FLAGS_gtest_death_test_style = "threadsafe";
 
 #ifdef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECKS
```
Comment (aprokop) [inline core/unit_test/TestViewCtorDimMatch.hpp:95]: Shouldn't it be `#ifndef` here and other places?

Comment (dalg24) [inline core/unit_test/TestViewCtorDimMatch.hpp:95]:
```suggestion
#ifndef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECKS
```

File: core/unit_test/TestViewCtorDimMatch.hpp [hunk 3]
```diff
@@ -119,6 +121,9 @@ TEST(TEST_CATEGORY_DEATH, view_construction_with_wrong_params_stat) {
...
```
Comment (dalg24) [inline core/unit_test/TestViewCtorDimMatch.hpp:123]:
```suggestion
#ifndef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECKS
```

File: core/unit_test/TestViewCtorDimMatch.hpp [hunk 4]
```diff
@@ -145,6 +149,9 @@ TEST(TEST_CATEGORY_DEATH, view_construction_with_wrong_params_mix) {
...
```
Comment (dalg24) [inline core/unit_test/TestViewCtorDimMatch.hpp:151]:
```suggestion
#ifndef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECKS
```

File: core/unit_test/TestViewCtorDimMatch.hpp [hunk 5]
```diff
@@ -174,6 +180,9 @@ TEST(TEST_CATEGORY_DEATH, view_construction_with_wrong_static_extents) {
...
```
Comment (dalg24) [inline core/unit_test/TestViewCtorDimMatch.hpp:182]:
```suggestion
#ifndef KOKKOS_ENABLE_DEBUG_BOUNDS_CHECKS
```

**Review by aprokop — APPROVED** [review]
